Laravel-whib/app/Http/Controllers/MapController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\MapPin;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Validation\ValidationException;
+
+class MapController extends Controller
+{
+    /**
+     * Display a listing of the map pins.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function index()
+    {
+        $mapPins = MapPin::all();
+
+        return response()->json(['map_pins' => $mapPins]);
+    }
+
+    /**
+     * Store a newly created map pin in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'pin_name' => 'required|string',
+            'description' => 'required|string',
+            'favourite' => 'required|bool',
+            'latitude' => 'required|numeric',
+            'longitude' => 'required|numeric',
+            'user_id' => "required|string",
+            'category' => "required|string",
+            // Add other validation rules as needed
+        ]);
+
+        $mapPin = MapPin::create($request->all());
+
+        return response()->json(['map_pin' => $mapPin], 201);
+    }
+
+    /**
+     * Display the specified map pin.
+     *
+     * @param  \App\Models\MapPin  $mapPin
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function show(MapPin $mapPin)
+    {
+        return response()->json(['map_pin' => $mapPin]);
+    }
+
+    /**
+     * Update the specified map pin in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \App\Models\MapPin  $mapPin
+     * @return \Illuminate\Http\JsonResponse
+     */
+
+    public function update(Request $request, MapPin $mapPin)
```

Comment:
Frontend should not pass PinMaps two times. Right now they would need to pass these parameters two times. Or maybe i am wrong idk.